quiche/src/lib.rs

```diff
@@ -7227,6 +7297,24 @@ impl<F: BufFactory> Connection<F> {
                             self.recovery_config.max_send_udp_payload_size,
                         );
                     }
+
+                    if ex_data.local_transport_params !=
+                        self.local_transport_params
+                    {
+                        if ex_data.local_transport_params.initial_max_streams_bidi !=
+                            self.local_transport_params
+                                .initial_max_streams_bidi
+                        {
```

Comment:
do we really need this if statement? Can we always just call `set_max_streams_bidi` and if they are the same its a noop.